src/__test__/class.test.ts

```diff
@@ -4,7 +4,7 @@ export class FooBar {
   }
 
   async bar(): Promise<string> {
-    return 'bar';
+    return await Promise.resolve('bar');
```

Comment:
I think the await is optional here (just `return Promise.resolve...` unless the value is used within the method)

Reply:
the await is not optional as the method is `async` there is no point in the method being `async` unless a `await` is used.

granted this is a very useless method :)

Reply:
my understanding is async only requires a promise to be returned, but it usually only makes sense to await when the value is used within a method.

Reply:
Its quite nitpicking but, async functions always return a promise, but they only need to be `async` if `await` is used.

I believe their is a performance penalty for using them so in the below example the second function is more correct. :shrug: 

```typescript
async foo() { return "bar" } 

foo() // Promise<"bar">

// More correct ?
fooProm() { return Promise.resolve("bar") }

fooProm() // Promise<"bar">

```

Reply:
I think `await Promise.resolve` is redundant. You could either `return await 'bar'`, or `return Promise.resolve('bar')` (and in this case not use `async` either). The consumer would need to await/.then it anyway.